graphqlapi/src/main/java/io/stargate/graphql/schema/fetchers/CassandraFetcher.java

```diff
@@ -25,8 +30,30 @@ public final ResultT get(DataFetchingEnvironment environment) throws Exception {
 
     String token = httpAwareContext.getAuthToken();
     StoredCredentials storedCredentials = authenticationService.validateToken(token);
-    DataStore dataStore = DataStore.create(persistence, storedCredentials.getRoleName());
 
+    ImmutableParameters.Builder parameters  = Parameters.builder();
+
+    if (environment.containsArgument("options")) {
+      Map<String, Object> options = environment.getArgument("options");
+      Object consistency = options.containsKey("consistency");
+      if (consistency != null) {
+        parameters.consistencyLevel(ConsistencyLevel.valueOf((String)consistency));
+      }
+      Object serialConsistency =  options.get("serialConsistency");
+      if (serialConsistency != null) {
+        parameters.serialConsistencyLevel(ConsistencyLevel.valueOf((String)options.get("serialConsistencyLevel")));
+      }
+      Object pageSize = options.get("pageSize");
+      if (pageSize != null) {
+        parameters.pageSize(Integer.valueOf((String)pageSize));
+      }
+      Object pageState = options.get("pageState");
+      if (pageSize != null) {
+        parameters.pagingState(PagingState.fromString((String)pageState).getRawPagingState());
+      }
+    }
+
+    DataStore dataStore = DataStore.create(persistence, storedCredentials.getRoleName(), parameters.build());
```

Comment:
Nit: you could avoid allocating the builder if there are no options.
```java
    Parameters parameters;
    if (environment.containsArgument("options")) {
      ImmutableParameters.Builder builder = Parameters.builder();
      ...
      parameters = builder.build();
    } else {
      parameters = Parameters.defaults();
    }
```

Reply:
Thanks, good suggestion.